projects/netgrif-components-core/src/lib/task/services/assign-policy.service.ts

```diff
@@ -49,6 +52,22 @@ export class AssignPolicyService extends TaskHandlingService {
      * @param afterAction the action that should be performed when the assign policy (and all following policies) finishes
      */
     public performAssignPolicy(taskOpened: boolean, afterAction: AfterAction = new AfterAction()): void {
+        if (!this._userService.isCurrentUserEmpty()) {
+            this.performAssign(taskOpened, afterAction);
+        } else {
+            race([
+                this._userService.anonymousUser$,
+                this._userService.user$
+            ]).pipe(take(1))
```

Comment:
@minop Sorry, that I was on sick leave last week, I have read the comments regarding the long-life subscription. Yes, I was aware with nature of `race` function, and you were right with the long-life subscription, but in my opinion `pipe` is not a proper solution, because the observables from `userService` (`anonymousUser$`, `user$`) can be updated multiple times and this with empty values too, so there is a possibility, that `take(1)` gets only empty user. 

A proper solution may be `takeUntil` or `takeWhile`, I will make proper investigation to this.

What do you think about it?